src/__swaps__/screens/Swap/resources/assets/userAssets.ts

```diff
@@ -222,3 +225,42 @@ export function useUserAssets<TSelectResult = UserAssetsResult>(
     staleTime: process.env.IS_TESTING === 'true' ? 0 : 1000,
   });
 }
+
+function getCachedUserAssets({
+  address,
+  currency,
+  testnetMode = false,
+}: {
+  address: Address;
+  currency: SupportedCurrencyKey;
+  testnetMode?: boolean;
+}) {
+  return queryClient.getQueryData<UserAssetsResult>(userAssetsQueryKey({ address, currency, testnetMode }));
+}
+
+const getNetworkNativeAssetUniqueId = (chainId: ChainId) => {
+  const network = getNetworkFromChainId(chainId);
+  const { nativeCurrency } = getNetworkObj(network);
+  const { mainnetAddress, address } = nativeCurrency;
+  const uniqueId = mainnetAddress ? `${mainnetAddress}_${ChainId.mainnet}` : `${address}_${chainId}`;
+  return uniqueId;
+};
+
+export function getUserNativeNetworkAsset(chainId: ChainId) {
+  const { accountAddress: currentAddress, nativeCurrency: currentCurrency, network: currentNetwork } = store.getState().settings;
+
+  const provider = getCachedProviderForNetwork(currentNetwork);
+  const providerUrl = provider?.connection?.url;
+  const connectedToHardhat = isHardHat(providerUrl);
```

Comment:
not sure if this landed or not yet, but christian wrote a `isHardhatConnected` function which does these three under the hood.